File: binary_search.js
const array_list = [1, 3, 5, 7, 9]


function binarySearch(arr, item) {
    let low = 0
    let high = arr.length - 1

    while (low <= high) {
        let mid = Math.floor((low + high) / 2)
        let guess = arr[mid]

        if(guess == item) {
            return mid
        }
        
        if(item < arr[mid]) {
            high = mid + 1
        } else {
            low = mid + 1
        }
    }

    return undefined
}

console.log(binarySearch(array_list, 3))
console.log(binarySearch(array_list, 10))

File: binary_search.py
my_list = ['apple', 'pineapple', 'orange', 'blueberry', 'graps']

def Binary_search (list, item):
    low = 0
    high = len(list) - 1

    while low <= high:
        mid = (low + high)
        guess = list[mid]

        if guess == item:
            return mid
        if guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None

print (Binary_search(my_list, 'blueberry'))
print (Binary_search(my_list, 'graps'))


'''my_list = 'Hello world'
print(len(my_list))'''